server/blob/blob-storage-strategy/src/test/java/org/apache/james/server/blob/deduplication/MinIOGenerationAwareBlobIdTest.java

```diff
@@ -0,0 +1,295 @@
+
+            BlobId actual = testee.parse(blobIdString);
+            assertThat(actual)
+                .isInstanceOfSatisfying(MinIOGenerationAwareBlobId.class, actualBlobId -> {
+                    SoftAssertions.assertSoftly(soft -> {
+                        soft.assertThat(actualBlobId.getFamily()).isEqualTo(0);
+                        soft.assertThat(actualBlobId.getGeneration()).isEqualTo(0L);
+                        soft.assertThat(actualBlobId.getDelegate()).isEqualTo(delegate.of(originalBlobId));
+                    });
+                });
+        }
+
+        @Test
+        void generationBlobIdShouldBeParsable() {
+            String originalBlobId = "abcdef";
+            String blobIdString = "12/126/" + delegate.of(originalBlobId).asString();
+
+            BlobId actual = testee.parse(blobIdString);
+            assertThat(actual)
+                .isInstanceOfSatisfying(MinIOGenerationAwareBlobId.class, actualBlobId -> {
+                    SoftAssertions.assertSoftly(soft -> {
+                        soft.assertThat(actualBlobId.getFamily()).isEqualTo(12);
+                        soft.assertThat(actualBlobId.getGeneration()).isEqualTo(126L);
+                        soft.assertThat(actualBlobId.getDelegate()).isEqualTo(delegate.of(originalBlobId));
+                    });
+                });
+        }
+
+        @Test
+        void wrappedBlobIdCanContainSeparator() {
+            String blobIdString = "12/126/ab/c";
+
+            BlobId actual = testee.parse(blobIdString);
+            assertThat(actual)
+                .isInstanceOfSatisfying(MinIOGenerationAwareBlobId.class, actualBlobId -> {
+                    SoftAssertions.assertSoftly(soft -> {
+                        soft.assertThat(actualBlobId.getFamily()).isEqualTo(12);
+                        soft.assertThat(actualBlobId.getGeneration()).isEqualTo(126L);
+                        soft.assertThat(actualBlobId.getDelegate()).isEqualTo(delegate.of("ab/c"));
+                    });
+                });
+        }
+
+        @ParameterizedTest
+        @ValueSource(strings = {
+            "abcdefgh",
+            "abcdefgh/",
+            "1/abcdefgh",
+            "1/2",
+            "1/2/",
+            "/abcdefgh"
+        })
+        void fromShouldFallbackWhenNotApplicable(String blobIdString) {
+            BlobId actual = testee.parse(blobIdString);
+            assertThat(actual)
+                .isInstanceOfSatisfying(GenerationAwareBlobId.class, actualBlobId -> {
+                    SoftAssertions.assertSoftly(soft -> {
+                        soft.assertThat(actualBlobId.getFamily()).isEqualTo(0);
+                        soft.assertThat(actualBlobId.getGeneration()).isEqualTo(0L);
+                        soft.assertThat(actualBlobId.getDelegate()).isEqualTo(delegate.parse(blobIdString));
+                    });
+                });
+        }
+
+        @Test
+        void fallbackGenerationAwareBlobIdShouldSuccess() {
```

Comment:
it fails
I want to confirm  https://github.com/apache/james-project/pull/2539#discussion_r1866984597 first

Reply:
No we do not need this fallback.